src/main/java/com/mraof/minestuck/command/ReviewDialogueCommand.java

```diff
@@ -0,0 +1,93 @@
+
+import java.util.Collection;
+
+public final class ReviewDialogueCommand
+{
+	public static final String INVALID_TYPE_KEY = "commands.minestuck.review_dialouge.invalid_type";
+	public static final String SUCCESS_KEY = "commands.minestuck.review_dialouge.success";
+	private static final SimpleCommandExceptionType ERROR_FAILED = new SimpleCommandExceptionType(Component.translatable("commands.summon.failed"));
+	private static final SimpleCommandExceptionType INVALID_TYPE = new SimpleCommandExceptionType(Component.translatable(INVALID_TYPE_KEY));
+	
+	public static void register(CommandDispatcher<CommandSourceStack> dispatcher, CommandBuildContext buildContext)
+	{
+		dispatcher.register(Commands.literal("review_dialogue").requires(source -> source.hasPermission(Commands.LEVEL_GAMEMASTERS))
+				.then(Commands.argument("entity", ResourceArgument.resource(buildContext, Registries.ENTITY_TYPE))
+						.suggests(MSSuggestionProviders.instance().DIALOGUE_ENTITY_TYPE)
+						.then(Commands.argument("category", new DialogueCategoryArgument())
+								.executes(context -> spawnDialogueEntities(context.getSource(),
+										ResourceArgument.getSummonableEntityType(context, "entity").value(),
+										DialogueCategoryArgument.getCategory(context, "category"))))));
+	}
+	
+	private static int spawnDialogueEntities(CommandSourceStack source, EntityType<?> entityType, RandomlySelectableDialogue.DialogueCategory category) throws CommandSyntaxException
+	{
+		Collection<Dialogue.SelectableDialogue> dialogueCollection = RandomlySelectableDialogue.instance(category).getAll();
+		
+		ServerLevel level = source.getLevel();
+		BlockPos pos = BlockPos.containing(source.getPosition());
+		
+		for(Dialogue.SelectableDialogue dialogue : dialogueCollection)
+		{
+			pos = pos.east(3);
+			level.removeBlock(pos, false);
+			level.setBlock(pos.below(), Blocks.BRICKS.defaultBlockState(), Block.UPDATE_ALL);
+			
+			Entity entity = entityType.spawn(level, pos, MobSpawnType.COMMAND);
+			if(entity == null)
+				throw ERROR_FAILED.create();
+			
+			entity.setNoGravity(true);
+			if(entity instanceof Mob mob)
+				mob.setNoAi(true);
+			
+			if(!(entity instanceof DialogueEntity dialogueEntity))
+				throw INVALID_TYPE.create();
+			
+			dialogueEntity.getDialogueComponent().setDialogue(dialogue.dialogueId(), true);
+			
+			BlockPos signPos = pos.north().below();
+			level.setBlock(signPos, Blocks.OAK_WALL_SIGN.defaultBlockState().setValue(WallSignBlock.FACING, Direction.NORTH), Block.UPDATE_ALL);
+			if(level.getBlockEntity(signPos) instanceof SignBlockEntity sign)
+			{
+				SignText signText = new SignText().setHasGlowingText(true).setColor(DyeColor.WHITE)
+						.setMessage(0, Component.literal(dialogue.dialogueId().getNamespace()));
+				String[] lines = dialogue.dialogueId().getPath().split("/", 3);
+				for(int i = 0; i < lines.length; i++)
+					signText = signText.setMessage(i + 1, Component.literal(lines[i]));
```

Comment:
I assume because they count as a single word, the names of long dialogues such as "a_little_lampshading" and "no_tower_treasure" get cut off

Reply:
A bit unfortunate. Maybe it'd be possible to switch to something other than signs, like attaching a name to the entity like a name tag. That said, I don't think it is a particularly important issue in this specific situation.